ci/test/actions-demo

```diff
@@ -297,3 +297,23 @@ EOF
 
 # test action runs sucessfully, cleanup action in not triggered
 popper run --on-failure cleanup
+
+
+# test prevention of double execution of actions
+cat <<EOF > main.workflow
+workflow "test-on-failure" {
+  resolves = ["test"]
+}
+
+action "test" {
+  uses = "actions/docker/cli@master"
+  args = ["verssion"]
+}
+
+action "cleanup" {
+  uses = "actions/bin/sh@master"
+  args = ["echo Cleaning up..."]
+}
+EOF
+
+popper run --on-failure cleanup
```

Comment:
can we grep the output to see if `Cleaning up...` is being printed? And similarly, in the workflow above where `--on-failure` should also be triggered?